src/Npgsql/Compatibility/CrateDB/CrateDBDatabaseInfo.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Npgsql.TypeMapping;
+using NpgsqlTypes;
+using System.Data;
+using Npgsql.TypeHandlers;
+using Npgsql.PostgresTypes;
+
+namespace Npgsql.Compatibility.CrateDB
+{
+    /// <summary>
+    /// Adapts type loading queries and type mappings for CrateDB.
+    /// </summary>
+    /// <remarks>
+    /// CrateDB contains support for the PostgreSQL wire protocol and it has a  
+    /// trimmed down pg_type table. But it does not have a pg_namespace-, pg_proc-,
+    /// pg_range- or pg_attribute-table. Further, the supported datatypes differ from PostgreSQL. 
+    /// See: https://crate.io/docs/crate/reference/en/latest/protocols/crate-postgres-sql.html for 
+    /// the main differences in implementation and dialect between CrateDB and PostgreSQL.
+    /// See: https://crate.io/docs/crate/reference/en/latest/protocols/postgres.html for more 
+    /// information about the support of the Postgres Wire Protocol in CrateDB.
+    /// </remarks>
+    class CrateDBDatabaseInfo : DatabaseInfo
```

Comment:
I need to think a bit more about how this is best implemented/architected. I'll get back to you soon on this.